app/domain/authentication/authn_jwt/identity_providers/create_identity_provider.rb

```diff
@@ -23,26 +23,32 @@ def call
 
         def create_identity_provider
           @logger.debug(LogMessages::Authentication::AuthnJwt::SelectingIdentityProviderInterface.new)
-          identity_provider = @identity_from_decoded_token_class.new(@authentication_parameters)
-          if identity_provider.identity_available?
+          identity_from_decoded_token_provider = @identity_from_decoded_token_class.new(@authentication_parameters)
```

Comment:
Please convert `identity_from_decoded_token_provider` & `identity_from_url_provider` to functions with memoizations